framework/src/wifi_manager/wifi_manager_info.c

```diff
@@ -0,0 +1,103 @@
+/****************************************************************************
+ *
+ * Copyright 2020 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <stdint.h>
+#include <tinyara/wifi/wifi_common.h>
+#include <wifi_manager/wifi_manager.h>
+#include "wifi_manager_state.h"
+#include "wifi_manager_info.h"
+
+struct _wifimgr_info {
+	char ssid[WIFIMGR_SSID_LEN + 1];             // SSID of Connected AP if mode is a station
+	char softap_ssid[WIFIMGR_SSID_LEN + 1];      // SoftAP SSID if mode is a soft ap
+	char mac_address[WIFIMGR_MACADDR_LEN];	   // MAC address of wifi interface
+	int rssi;                  // It is only used for a station mode
+	wifimgr_state_e state;
+	wifimgr_state_e prev_state;
+};
+typedef struct _wifimgr_info _wifimgr_info_s;
+
+static _wifimgr_info_s g_manager_info = {{0}, {0}, {0}, 0};
+
+
+int wifimgr_get_info(int flag, wifimgr_info_msg_s *info)
```

Comment:
A null-check for `*info` would be useful to have, just in case things go wrong. Please ignore if not necessary.

Reply:
it's already checked to caller